### PR TITLE
migrating ssh config to team config

### PR DESCRIPTION
**Pull request by jace-roell**
**What It Does**
Adds capability to add new ssh configs
 - migrating from existing SSH config 
 - building config from input SSH command
 
**Features**
- allows for selection of authentication and prompting for secure credentials
- prompts for confirmation if name matches an existing profile
- authentication method quickpick order is dependent on if IdentityFile/PrivateKey is passed from config
- Private key file picker will open to .ssh directory.  if it cannot be found, open to homedir
- once a config is migrated from ssh to team config, it will no longer be placed in the migrate area based on the hostname already being utilized in the config. This may contain an edge case where a user has two configurations using the same host name for seperate purposes and once one is migrated the other will not be in the list. 

**How to test**
If you do not already have an SSH config , create one, default location:  `/<user>/.ssh/config`
Example:
```
Host Test1
  HostName hello.com
  User user
Host test@host.com
  HostName host.com
  user zoweuser
  Port 1234
  IdentityFile /Users/user/.ssh/id_rsa
  ConnectTimeout 50

```
- Start extension
- Run `Zowe-SSH: Restart Zowe Server on Host...` in command palette.
- Add a new config based on ssh conection string or migrate from existing SSH config

**Future Enhancement Story**
Automatically attempt SSH connection with credentials before prompting for more information that may not be necessary

**Review Checklist**
I certify that I have:
- [x] tested my changes
- [ ] added/updated automated tests
- [ ] updated the changelog
- [x] followed the [contribution guidelines](https://github.com/zowe/zowe-cli/blob/master/CONTRIBUTING.md)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -32,8 +32,9 @@
   "devDependencies": {
     "@biomejs/biome": "^1.9.4",
     "@zowe/secrets-for-zowe-sdk": "^8.10.4",
+    "tsx": "^4.19.2",
+    "ssh-config": "^5.0.3",
     "husky": "^9.1.7",
     "rimraf": "^6.0.1",
```
Comment (t1m0thyj) [inline package.json:38]: Please remove the trailing comma - `npm install` emits an error about invalid syntax

File: package.json [hunk 2]
```diff
@@ -32,8 +32,9 @@
   "devDependencies": {
     "@biomejs/biome": "^1.9.4",
     "@zowe/secrets-for-zowe-sdk": "^8.10.4",
+    "tsx": "^4.19.2",
+    "ssh-config": "^5.0.3",
```
Comment (t1m0thyj) [inline package.json:36]: We want the "ssh-config" dep to be a prod dep in the package.json of the package using it (currently `packages/vsce`), rather than a dev dep at the top-level.

File: packages/vsce/src/SshConfigUtils.ts [hunk 3]
```diff
@@ -23,27 +32,105 @@ export class SshConfigUtils {
...
+                selectedProfile = await SshConfigUtils.getNewProfileName(selectedProfile);
+                selectedProfile = await SshConfigUtils.promptForAuth(selectedProfile);
+            }
+        }
+        else if (result != null) {
             return sshProfiles.find((prof) => prof.name === result.label);
         }
     }
 
+    private static async migrateSshConfig(): Promise<sshConfigExt[]> {
```
Comment (t1m0thyj) [inline packages/vsce/src/SshConfigUtils.ts:82]: Should we move this method to the SDK package and make it a public method there so that it can be invoked by extenders outside of the VSCE?

File: packages/vsce/src/SshConfigUtils.ts [hunk 4]
```diff
@@ -58,7 +145,234 @@ export class SshConfigUtils {
...
+                for (const algo of ["id_ed25519", "id_rsa"]) {
+                    const tempPath = path.resolve(homedir(), ".ssh", algo);
+                    if (fs.existsSync(tempPath)) {
+                        privateKeyPath = path.resolve(homedir(), ".ssh", algo);
+                        break;
+                    }
+                }
+                if (SshProfile.privateKey == null) {
+                    throw Error("Failed to discover an ssh private key inside `~/.ssh`.");
+                }
```
Comment (t1m0thyj) [inline packages/vsce/src/SshConfigUtils.ts:232]: Should we move this to a public method like `findPrivateKey` in the SDK package?

**Review by t1m0thyj — COMMENTED** [review]
Looks pretty good, thanks @jace-roell! Left a few comments - regarding moving functions to the SDK, feel free to create a separate issue to do that in the future.

**Review by zFernand0 — APPROVED** [review]
LGTM! 😋 
---
This is really impressive 🤯
